Extensions/BBText/bbtextruntimeobject.js

```diff
@@ -0,0 +1,206 @@
+  /** @type string */
+  this._family = objectData.content.family;
+  /** @type number */
+  this._size = objectData.content.size;
+  /** @type number */
+  this._wrappingWidth = objectData.content.width;
+  /** @type string */
+  this._align = objectData.content.align;
+
+  if (this._renderer)
+    gdjs.BBTextRuntimeObjectRenderer.call(this._renderer, this, runtimeScene);
+  else
+    this._renderer = new gdjs.BBTextRuntimeObjectRenderer(this, runtimeScene);
+
+  // *ALWAYS* call `this.onCreated()` at the very end of your object constructor.
+  this.onCreated();
+};
+
+gdjs.BBTextRuntimeObject.prototype = Object.create(
+  gdjs.RuntimeObject.prototype
+);
+gdjs.BBTextRuntimeObject.thisIsARuntimeObjectConstructor = 'BBText::BBText';
+
+gdjs.BBTextRuntimeObject.prototype.getRendererObject = function() {
+  return this._renderer.getRendererObject();
+};
+
+/**
+ * Initialize the extra parameters that could be set for an instance.
+ * @private
+ */
+gdjs.BBTextRuntimeObject.prototype.extraInitializationFromInitialInstance = function(
+  initialInstanceData
+) {
+  if (initialInstanceData.customSize) {
+    this.setWidth(initialInstanceData.width);
+    this.setWrappingWidth(initialInstanceData.width);
+  }
+};
+
+gdjs.BBTextRuntimeObject.prototype.onDestroyFromScene = function(runtimeScene) {
+  gdjs.RuntimeObject.prototype.onDestroyFromScene.call(this, runtimeScene);
+
+  this._renderer.onDestroy();
+};
+
+gdjs.BBTextRuntimeObject.prototype.update = function(runtimeScene) {
+  if (this._renderer) this._renderer.ensureUpToDate();
+};
+
+/**
+ * Set/Get BBText base style properties
+ */
+gdjs.BBTextRuntimeObject.prototype.setBBText = function(text) {
+  this._text = text;
+  this._renderer.updateText();
+};
+
+gdjs.BBTextRuntimeObject.prototype.getBBText = function(text) {
+  return this._text;
+};
+
+gdjs.BBTextRuntimeObject.prototype.setColor = function(value) {
+  const splitValue = value.split(';');
+  const hexColor = `#${gdjs.rgbToHex(
+    parseInt(splitValue[0]),
+    parseInt(splitValue[1]),
+    parseInt(splitValue[2])
+  )}`;
+  this._color = hexColor;
+  this._renderer.updateColor();
+  this.update();
+};
+
+gdjs.BBTextRuntimeObject.prototype.getColor = function() {
+  return this._color;
+};
+
+gdjs.BBTextRuntimeObject.prototype.setFontSize = function(value) {
+  this._size = `${value}px`;
+  this._renderer.updateFontSize();
+  this.update();
+};
+
+gdjs.BBTextRuntimeObject.prototype.getFontSize = function() {
+  return this._size;
+};
+
+gdjs.BBTextRuntimeObject.prototype.setFontFamily = function(value) {
+  this._family = value;
+  this._renderer.updateFontFamily();
+  this.update();
+};
+
+gdjs.BBTextRuntimeObject.prototype.getFontFamily = function() {
+  return this._family;
+};
+
+gdjs.BBTextRuntimeObject.prototype.setAlignment = function(value) {
+  this._align = value;
+  this._renderer.updateAlignment();
+  this.update();
+};
+
+gdjs.BBTextRuntimeObject.prototype.getAlignment = function() {
+  return (this._align = value);
```

Comment:
Careful here you're setting _align to undefined or crashing the game because of a undefined variable.

Reply:
oops, sorry some bad late night copy pasta there. I must have taken it from the get function and forgotten to delete the = value  part